obsreport/obsreport_receiver.go

```diff
@@ -61,7 +66,7 @@ func NewReceiver(cfg ReceiverSettings) *Receiver {
 			tag.Upsert(obsmetrics.TagKeyReceiver, cfg.ReceiverID.String(), tag.WithTTL(tag.TTLNoPropagation)),
 			tag.Upsert(obsmetrics.TagKeyTransport, cfg.Transport, tag.WithTTL(tag.TTLNoPropagation)),
 		},
-		tracer: cfg.ReceiverCreateSettings.TracerProvider.Tracer(cfg.ReceiverID.String()),
+		tracer: traceProvider.Tracer(cfg.ReceiverID.String()),
```

Comment:
If `cfg.ReceiverCreateSettings.TracerProvider` is nil, this part fails with a fatal.